docs/design/2021-03-09-dynamic-privileges.md

```diff
@@ -0,0 +1,282 @@
+# Proposal:
+
+- Author(s):     [morgo](https://github.com/morgo)
+- Last updated:  March 09, 2021
+- Discussion at: N/A
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+    * [Compatibility Tests](#compatibility-tests)
+    * [Benchmark Tests](#benchmark-tests)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+This document was created to discuss the design of Dynamic Privileges. It is intended to be implemented in combination with Security Enhanced Mode, but there no interdependencies between the two features.
+
+## Motivation or Background
```

Comment:
I think the following "Audit Admin" is the example: suppose we want some administrators to be able to change audit settings, but before Dynamic Priviligies, the only way is to grant "SUPER" because the privilige is too coarse.

BTW, IMHO the naming of 'Dyanmic Priviligies' is kind of misleading, the core intention of it is make the priviligies 'fine grained' and 'dynamic'.

Reply:
I think it is maybe an implementation detail, but I don't think it is misleading. In my original proposal I started it with a section called 'terminology', but this is not part of the new standard template, so I removed it.

The implementation detail is more relevant to MySQL, because it has quite a number of plugins. It is less relevant to TiDB, but it is refered to as [dynamic privileges](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#privileges-provided-dynamic) in the documentation, so I don't recommend changing it.

I am happy to add an example to make it clearer though.

Reply:
I have added a paste of a testcase. There are additional examples in the linked MySQL worklog WL#8131, which is in the first sentence.